optuna/pruners/_wilcoxon.py

```diff
@@ -58,22 +58,22 @@ def eval_func(param, input_):
 
 
             input_data = np.linspace(-1, 1, 100)
-
-            # It is recommended to shuffle the input data once before optimization.
-            np.random.shuffle(input_data)
+            rng = np.random.default_rng()
 
 
             def objective(trial):
-                s = 0.0
-                for i in range(len(input_data)):
+                # In each trial, it is recommended to shuffle the order in which data is processed.
+                ordering = rng.permutation(range(len(input_data)))
+                s = []
+                for i in ordering:
                     param = trial.suggest_float("param", -1, 1)
                     loss = eval_func(param, input_data[i])
                     trial.report(loss, i)
-                    s += loss
+                    s.append(loss)
                     if trial.should_prune():
                         raise optuna.TrialPruned()
```

Comment:
```suggestion
                        return sum(s) / len(s)
```
Could you also change the docstring to explain that this is necessary, because this is different from other pruners?

Reply:
I agree that the suggested technique is a beneficial workaround. However, I believe it might be too advanced for use in a docstring example. In addition, it is not necessary; for example, it has no effect with GridSampler.
I added an comment in the code and wrote a separate note.

Reply:
If you add a note in the docstring, why not demonstrate that in the example?
This workaround is actually necessary for all samplers except for `GridSampler`, `BruteForceSampler`, `QMCSampler` and `RandomSampler` I think.

Reply:
I see; I modified the example so that `return sum(s) / len(s)`.